crypto/threads_win.c

```diff
@@ -154,7 +156,8 @@ CRYPTO_RCU_LOCK *ossl_rcu_lock_new(int num_writers, OSSL_LIB_CTX *ctx)
         || new->prior_signal == NULL
         || new->write_lock == NULL
         || new->alloc_lock == NULL
-        || new->prior_lock == NULL) {
+        || new->prior_lock == NULL
+        || new->rw_lock == NULL) {
```

Comment:
you also need to free rw_lock here.  e.g if you successfully allocate rw_lock but prior_lock returns null, you will still get  a leak.  Its ok if you call an objects free function with a NULL pointer, they all check for that

Reply:
I added a free call in case of an error.